### ISSUE TITLE
SCM allow commit author to be define in settings.py

### ISSUE THREAD

**Issue by pyther** [cobbler/cobbler#1377]
It is undesirable to have a commit author of root@localhost.localdomain. A user-defined variable in settings.py would be useful.

Proof of concept patch:
```diff
--- cobbler/modules/scm_track.py.orig   2014-12-12 14:37:33.908482842 -0500
+++ cobbler/modules/scm_track.py        2014-12-12 14:42:50.450071316 -0500
@@ -46,11 +46,15 @@ def run(api,args,logger):
     settings = api.settings()
     scm_track_enabled  = str(settings.scm_track_enabled).lower()
     mode = str(settings.scm_track_mode).lower()
+    author = str(settings.scm_track_author)
+    if not author:
+        author = 'cobbler <root@localhost.localdomain>'
+
 
     if scm_track_enabled not in [ "y", "yes", "1", "true" ]:
        # feature disabled
        return 0
-   
+ 
     if mode == "git":
 
        old_dir = os.getcwd()
@@ -66,7 +70,7 @@ def run(api,args,logger):
        rc = utils.subprocess_call(logger,"git add --all config",shell=True)
        rc = utils.subprocess_call(logger,"git add --all kickstarts",shell=True)
        rc = utils.subprocess_call(logger,"git add --all snippets",shell=True)
-       rc = utils.subprocess_call(logger,"git commit -m 'API update' --author 'cobbler <root@localhost.localdomain>'",shell=True)
+       rc = utils.subprocess_call(logger,"git commit -m 'API update' --author '{0}'".format(author),shell=True)
 
        os.chdir(old_dir)
        return 0
@@ -86,7 +90,7 @@ def run(api,args,logger):
         rc = utils.subprocess_call(logger,"hg add config",shell=True)
         rc = utils.subprocess_call(logger,"hg add kickstarts",shell=True)
         rc = utils.subprocess_call(logger,"hg add snippets",shell=True)
-        rc = utils.subprocess_call(logger,"hg commit -m 'API update' --user 'cobbler <root@localhost.localdomain>'",shell=True)
+        rc = utils.subprocess_call(logger,"hg commit -m 'API update' --user '{0}'".format(author),shell=True)
 
         os.chdir(old_dir)
         return 0
```

In settings.py
`scm_track_author: 'cobbler@example.domain.net'`

**Comment by jmaas** [comment]
Indeed nice improve, Pyther can you send a pull request? That way you get the proper credits as well ;)

**Comment by bstrdsmkr** [comment]
Even better would be if the user triggering the Sync could be captured and used (at least in the web ui) so we can better audit who did what.

**Comment by jmaas** [comment]
Implemented in master (3.0) and 2.8 development branch.
